api/src/main/java/com/datastrato/gravitino/authorization/SecurableObjects.java

```diff
@@ -29,7 +33,41 @@ public static SecurableObject of(String... names) {
       throw new IllegalArgumentException("Cannot create a securable object with no names");
     }
 
+    if (type == null) {
+      throw new IllegalArgumentException("Cannot create a securable object with no type");
+    }
+
+    if (names.length > 3) {
+      throw new IllegalArgumentException(
+          "Cannot create a securable object with the name length which is greater than 3");
+    }
+
+    List<SecurableObjectType> typeInDifferentLevels =
+        Lists.newArrayList(SecurableObjectType.CATALOG, SecurableObjectType.SCHEMA);
+
+    if (names.length == 1 && type != SecurableObjectType.CATALOG) {
```

Comment:
For metalake, the length should also be 1?

Reply:
I will add some test code.